internal/common/conversions.go

```diff
@@ -1,10 +1,21 @@
 package common
 
 import (
+	"strings"
+
 	"github.com/go-openapi/strfmt"
+	"github.com/go-openapi/swag"
+	hiveext "github.com/openshift/assisted-service/api/hiveextension/v1beta1"
 	"github.com/openshift/assisted-service/models"
 )
 
+func PlatformTypeToPlatform(platformType hiveext.PlatformType) *models.Platform {
+	pType := strings.ToLower(string(platformType))
+	platform := &models.Platform{Type: PlatformTypePtr(models.PlatformType(pType))}
+	platform.IsExternal = swag.Bool(*platform.Type == models.PlatformTypeOci)
+	return platform
+}
```

Comment:
maybe make it more generic?

```suggestion
func PlatformTypeToPlatform[T hiveext.PlatformType | models.PlatformType](platformType T) *models.Platform {
	pType := strings.ToLower(string(platformType))
	platform := &models.Platform{Type: PlatformTypePtr(models.PlatformType(pType))}
	platform.IsExternal = swag.Bool(*platform.Type == models.PlatformTypeOci)
	return platform
}
```

Reply:
I don't think `PlatformTypeToPlatform` should accept the two types unless we actually use it for both purposes in the code.

Reply:
We do that same logic on multiple places. See almost every platform set on `test_inventory` (and probably more).
If you don't think that it's necessary I'll not insist on it

Reply:
I'm just not sure I am getting you.
This is a new conversion function I've added with this PR, and it is being used several times in `pkg/webhooks/hiveextension/v1beta1/agentclusterinstall_validation_hook.go`, but not in other places.

What am I missing?